### ISSUE TITLE
False-positive RedundantCondition: explode may return empty array

### ISSUE THREAD

**Issue by ddebin** [vimeo/psalm#1952]
```php
<?php

$elements = explode('_', '', -1);
$element = array_shift($elements);
assert(null !== $element);
```

https://psalm.dev/r/514b812a52

Expected: no errors
Actual: `ERROR: RedundantCondition - 5:8 - string can never contain null`

**Comment by muglug** [comment]
Fixed in https://github.com/vimeo/psalm/commit/59dd6393f51bfa21cb9ffb2f95fc45bf9296d7bf - thanks, didn't think about the third arg
